### ISSUE TITLE
Downloading file returns 403 even though it's accessible in the browser

### ISSUE THREAD

**Issue by ror3d** [JabRef/jabref#11063]
### JabRef version

5.12 (latest release)

### Operating system

Windows

### Details on version and operating system

Windows 11

### Checked with the latest development build (copy version output from About dialog)

- [X] I made a backup of my libraries before testing the latest development version.
- [X] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

1. Create a new entry
2. On the General tab, on the File section, choose the third button on the right to download from a link
3. Use the link https://royalsocietypublishing.org/doi/pdf/10.1098/rspa.2020.0351
4. Notice that a 403 error is reported
5. Try to access this file through the browser, notice that it lets you view the PDF file

This seems to happen with multiple, if not all, files from the royalsocietypublishing.org domain.

I have tried accessing the domain through the browser and spoofing the user agent used by JabRef and it let me access it, so that might not be the underlying cause (would be good to double-check though).

**Comment by Siedlerchr** [comment]
Yes, this unfortunately often happens if the site is using cloudflare or something similar. It's reyling on some javascript stuff that gets injected which we cannot do.
The overall idea is to replace the current fetcher infrastructure with a browser based integration

**Comment by ror3d** [comment]
I see something of the like if I run `curl https://royalsocietypublishing.org/doi/pdf/10.1098/rspa.2020.0351` for example, but I don't understand why it responds with a 403 - Forbidden instead for JabRef.

It's not really much of a problem, but compared to the other ways publishers avoid such downloads, this one is the one I understand the least.

**Comment by koppor** [comment]
Maybe, our user string is not good enough. Maybe our http behavior is strange. (Java vs. curl)

The "idea" is to play with https://github.com/JetBrains/jcef. https://github.com/chromiumembedded/java-cef/blob/master/java/tests/detailed/handler/RequestHandler.java looks interesting.

**Comment by chrisdenman** [comment]
See [#12864](https://github.com/JabRef/jabref/issues/12864)
